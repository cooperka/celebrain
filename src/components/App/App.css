.App {
  display: flex;
  flex-direction: column;
  text-align: center;
  background-color: #222;
}

.main-header {
  margin: 2em;
  margin-bottom: 0;
  color: white;
}

.main-logo {
  height: 8em;
  animation: main-logo-spin infinite 20s linear;
}

.main-title {
  margin: 1rem;
  font-size: 5em;
}

.main-subtitle {
  margin: 1rem;
  font-size: 2.2em;
}

.social-links {
  margin: 2em;
  margin-top: 0;
  padding: 0;
  list-style: none;
}

.social-link {
  margin: .5rem;
  padding: .5rem;
  font-size: 1.6em;
}

@keyframes main-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
