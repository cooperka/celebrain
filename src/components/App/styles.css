.App {
  display: flex;
  flex-direction: column;
  text-align: center;
  color: rgba(0, 0, 0, 0.87);
  margin: 0 12px;
}

.page {
  min-height: 30rem;
  margin: 4rem 0;
  display: flex;
  flex-direction: column;
  font-weight: 500;
}

.section {
  margin: 1rem auto;
  max-width: 55rem;
}

.title {
  margin: 1rem;
  font-size: 5rem;
}

.subtitle {
  margin: 1rem auto;
  font-size: 2.2rem;
  max-width: 44rem;
}
