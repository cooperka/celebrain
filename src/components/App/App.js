import React, { Component } from 'react';

import profilePhoto from '../../images/profile-2017-web@800.jpg';

import './App.css';

const SOCIAL_LINKS = [
  {
    name: 'Facebook',
    url: 'https://www.facebook.com/cooperka',
    icon: require('../../images/Facebook-color.svg'),
  },
  {
    name: 'LinkedIn',
    url: 'https://www.linkedin.com/in/cooperka92',
    icon: require('../../images/LinkedIn-color.svg'),
  },
  {
    name: 'GitHub',
    url: 'https://github.com/cooperka',
    icon: require('../../images/Github-color.svg'),
  },
  {
    name: 'Medium',
    url: 'https://medium.com/@cooperka',
    icon: require('../../images/Medium-color.svg'),
  },
  {
    name: 'Tumblr',
    url: 'https://mentalmorsels.tumblr.com/',
    icon: require('../../images/Tumblr-color.svg'),
  },
];

class App extends Component {

  renderMainHeader() {
    return (
      <div className="main-header">
        <img className="profile-photo" src={profilePhoto} alt="profile" />
        <div className="main-title">Kevin Cooper</div>
        <div className="main-subtitle">Software Engineer</div>
      </div>
    );
  }

  renderSocialLinks() {
    return (
      <ul className="social-links">
        {SOCIAL_LINKS.map(this.renderSocialLink)}
      </ul>
    );
  }

  renderSocialLink({ name, url, icon }, index) {
    return (
      <li className="social-link" key={index}>
        <a className="social-link" href={url}>
          <img className="social-icon" src={icon} alt={name} />
        </a>
      </li>
    );
  }

  render() {
    return (
      <div className="App">
        {this.renderMainHeader()}
        {this.renderSocialLinks()}
      </div>
    );
  }

}

export default App;
