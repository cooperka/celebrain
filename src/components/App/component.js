// @flow

import React, { Component } from 'react';

import logo from '../../images/full-logo.svg';
import SOCIAL_LINKS from '../../socialLinks';

import SocialLink from '../SocialLink';

import './styles.css';

type Props = {};

class App extends Component<Props> {

  renderMainHeader() {
    return (
      <div className="main-header">
        <img className="company-logo" src={logo} alt="Company logo" />
        <div className="main-subtitle">Lovable. Simple. Refined.</div>
      </div>
    );
  }

  renderSocialLinks() {
    return (
      <ul className="social-links">
        {SOCIAL_LINKS.map(
          (data, index) =>
            // eslint-disable-next-line react/no-array-index-key
            <SocialLink key={index} name={data.name} url={data.url} icon={data.icon} iconHover={data.iconHover} />,
        )}
      </ul>
    );
  }

  render() {
    return (
      <div className="App">
        {this.renderMainHeader()}
        {this.renderSocialLinks()}
      </div>
    );
  }

}

export default App;
