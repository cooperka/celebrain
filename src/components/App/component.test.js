// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import App from './component';

jest.mock('../SocialLink/component', () => 'SocialLink');

it('renders', () => {
  const tree = renderer.create(<App />);
  expect(tree.toJSON()).toMatchSnapshot();
});
