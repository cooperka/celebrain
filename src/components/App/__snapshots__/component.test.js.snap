// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="App"
>
  <div
    className="nav-bar material z-depth-1"
  >
    <img
      alt="Company logo"
      className="company-logo"
      src="logo.svg"
    />
  </div>
  <div
    className="main-header"
  >
    <div
      className="main-subtitle"
    >
      Lovable. Simple. Refined.
    </div>
    <div
      className="main-subtitle"
    >
      Full website coming soon. 
      <a
        href="mailto:kevin@lsrfocus.com"
      >
        Contact us
      </a>
       for inquiries.
    </div>
  </div>
  <ul
    className="social-links"
  >
    <SocialLink
      icon="Github-black.svg"
      iconHover="Github-color.svg"
      name="GitHub"
      url="https://github.com/lsrfocus"
    />
    <SocialLink
      icon="LinkedIn-black.svg"
      iconHover="LinkedIn-color.svg"
      name="LinkedIn"
      url="https://www.linkedin.com/company/18449687"
    />
    <SocialLink
      icon="Facebook-black.svg"
      iconHover="Facebook-color.svg"
      name="Facebook"
      url="https://www.facebook.com/LsrFocus"
    />
    <SocialLink
      icon="Twitter-black.svg"
      iconHover="Twitter-color.svg"
      name="Twitter"
      url="https://twitter.com/LsrFocus"
    />
  </ul>
</div>
`;
